processor/k8sprocessor/kube/owner.go

```diff
@@ -127,6 +131,40 @@ func newOwnerProvider(
 	return &ownerCache, nil
 }
 
+func (op *OwnerCache) upsertNamespace(obj interface{}) {
+	namespace := obj.(api_v1.Namespace)
+	op.cacheMutex.Lock()
+	op.namespaces[namespace.Name] = &namespace
+	op.cacheMutex.Unlock()
+}
+
+func (op *OwnerCache) deleteNamespace(obj interface{}) {
+	namespace := obj.(api_v1.Namespace)
+	op.cacheMutex.Lock()
+	delete(op.namespaces, namespace.Name)
+	op.cacheMutex.Unlock()
```

Comment:
There might be very little risk here and maybe not worth the trouble, but I see a lot of the following pattern when working with locks:

```
op.cacheMutex.Lock()
defer op.cacheMutex.Unlock()
....
```

Reply:
👍